### ISSUE TITLE
`Store.delete` has inconsistent implementations

### ISSUE THREAD

**Issue by d-v-b** [zarr-developers/zarr-python#2451]
`LocalStore.delete` removes entire directory trees, i.e. many objects:

https://github.com/zarr-developers/zarr-python/blob/4c3081c9b4678d1de438e0d2ee8e0fa368a61f17/src/zarr/storage/local.py#L206-L213

`MemoryStore.delete` removes single keys, i.e. just one object:

https://github.com/zarr-developers/zarr-python/blob/4c3081c9b4678d1de438e0d2ee8e0fa368a61f17/src/zarr/storage/memory.py#L134-L140

We should pick one of the two options. The fact that we got this far with such skew also means that our store tests should be made more extensive.

**Comment by jhamman** [comment]
Good catch! `Store.delete` should be for a single key. In https://github.com/zarr-developers/zarr-python/pull/2430, I introduce `delete_dir` for removing all objects below a node in a hierarchy.
